### PR TITLE
Feature/admin module

### PR DESCRIPTION
**Pull request by NurmukhammadMakhmudov**

### CODE REVIEW

File: src/app/models/admin-profile/admin-dialog/admin-dialog.component.ts [hunk 1]
```diff
@@ -1,29 +1,27 @@
...
 })
 export class AdminDialogComponent implements OnInit {
-  coaches = MOCK_COACH;
-
   coachControl = new FormControl('', Validators.required);
 
   constructor(
     public dialogRef: MatDialogRef<AdminDialogComponent>,
-    @Inject(MAT_DIALOG_DATA) public data: TestData,
+    @Inject(MAT_DIALOG_DATA) public data: any,
```
Comment (ExxZzyy) [inline src/app/models/admin-profile/admin-dialog/admin-dialog.component.ts:18]: Don't use any

File: src/mocks/admin-profile-utils.mock.ts [hunk 2]
```diff
@@ -1,68 +1,43 @@
 export interface CoachData {
```
Comment (ExxZzyy) [inline src/mocks/admin-profile-utils.mock.ts:1]: Move this interfaces into src/app/interfaces

File: src/app/models/admin-profile/admin-table/table.component.scss [hunk 3]
```diff
@@ -5,6 +5,12 @@ cdk-virtual-scroll-viewport {
   margin-left: 8vw;
 }
 
+mat-tab-group {
+  justify-self: center;
+  max-width: 100%;
+  margin: 0 50px;
```
Comment (tsiananel) [inline src/app/models/admin-profile/admin-table/table.component.scss:11]: Let use rem

File: src/app/models/admin-profile/admin-table/table.component.ts [hunk 4]
```diff
@@ -1,47 +1,79 @@
-import { Component, Input, OnInit, ViewChild } from '@angular/core';
+import { Component, OnInit, ViewChild } from '@angular/core';
 import { MatDialog } from '@angular/material/dialog';
-import { MatTable, MatTableDataSource } from '@angular/material/table';
-import { HttpClient } from '@angular/common/http';
-import { AdminTestTabs, TestData } from '../../../../mocks/admin-profile-utils.mock';
+import { MatTable } from '@angular/material/table';
```
Comment (tsiananel) [inline src/app/models/admin-profile/admin-table/table.component.ts:3]: All material components move to angular-material-common.module

**Review by tsiananel — CHANGES_REQUESTED** [review]

File: src/app/models/admin-profile/admin-table/table.component.ts [hunk 5]
```diff
@@ -1,47 +1,79 @@
...
 
-  @Input() displaedColums: string[] = [];
+  displaedColumshasCoach = ['Position', 'Level', 'Date', 'Coach', 'Button'];
 
-  constructor(
-    public dialog: MatDialog,
-    private readonly http: HttpClient,
-    private service: ServiceComponent,
-  ) {}
+  constructor(public dialog: MatDialog, private service: AdminTableStoreService) {}
```
Comment (shuliakleonid) [inline src/app/models/admin-profile/admin-table/table.component.ts:43]: I think we have double code and better do this with enum object

File: src/app/models/admin-profile/admin-table/table.component.ts [hunk 6]
```diff
@@ -1,47 +1,79 @@
...
 
   @ViewChild(MatTable) table!: MatTable<any>;
 
-  ngOnInit(): void {}
+  ngOnInit(): void {
+    this.service.getAssignedTestData().subscribe((assignedData) => {
+      console.log(assignedData);
+      this.assignedData = assignedData.results;
```
Comment (shuliakleonid) [inline src/app/models/admin-profile/admin-table/table.component.ts:50]: remove console.log

File: src/app/services/adminTableData.service.ts [hunk 7]
```diff
@@ -1,25 +1,42 @@
...
-    console.log(
-      this.http.get<TestData[]>('http://elevel-001-site1.btempurl.com/api/Test/forAdmin'),
+  options = {
+    headers: new HttpHeaders({ 'Content-Type': 'application/json' }),
+  };
+
+  getAssignedAdminTests() {
+    return this.http.get<ServiceTestData>(
+      'http://elevel-001-site1.btempurl.com/api/Test/forAdmin?IsAssigned=true',
     );
```
Comment (shuliakleonid) [inline src/app/services/adminTableData.service.ts:23]: we have variable for URL address

File: src/app/services/adminTableData.service.ts [hunk 8]
```diff
@@ -1,25 +1,42 @@
...
   }
 
-  passAdminTest(test: TestData) {
-    return this.http.post<TestData[]>(
-      'http://elevel-001-site1.btempurl.com/api/Test/forAdmin',
-      test,
+  getNotAssignedAdminTests() {
+    return this.http.get<ServiceTestData>('http://elevel-001-site1.btempurl.com/api/Test/forAdmin');
+  }
+
```
Comment (shuliakleonid) [inline src/app/services/adminTableData.service.ts:29]: use const BASE_API_URL

File: src/app/services/adminTableData.service.ts [hunk 9]
```diff
@@ -1,25 +1,42 @@
...
+  getAdminCoaches() {
+    return this.http.get<ServiceCoachData>('http://elevel-001-site1.btempurl.com/api/User/coaches');
+  }
+
+  updateCoachTest(data: UpdateCoachesData, testID: number) {
+    console.log(`http://elevel-001-site1.btempurl.com/api/Test/${testID}assignForCoach`);
+    return this.http.put<TestData>(
+      `http://elevel-001-site1.btempurl.com/api/Test/${testID}/assignForCoach`,
+      data,
```
Comment (shuliakleonid) [inline src/app/services/adminTableData.service.ts:38]: same

File: src/app/services/store/adminTableStore.service.ts [hunk 10]
```diff
@@ -8,11 +9,20 @@ import { AdminHttpService } from '../adminTableData.service';
...
+  }
+
+  getCoachData() {
+    return this.adminHttpService.getAdminCoaches();
+  }
+
+  updateTestData(data: UpdateCoachesData, testID: number) {
+    console.log(data);
+    this.adminHttpService.updateCoachTest(data, testID).subscribe((resp) => console.log(resp));
   }
```
Comment (shuliakleonid) [inline src/app/services/store/adminTableStore.service.ts:27]: remove console.log

**Review by shuliakleonid — CHANGES_REQUESTED** [review]

**Review by RomPavel — APPROVED** [review]

**Pull request merged**